bigtreecms/BigTree-CMS
# Sitemap

It would be nice to not generate sitemp.xml on-the-fly in order to speed up the response time of `/sitemap.xml` page. Rendering of huge sitemap with several thousands of links can take around 10s. That's is not cool.
So I am suggesting to not generate the sitemap.xml on-the-fly, but save it instead and we can update it in the background with the cron job. I don't think sitemap is a thing that is updated every second or minute, that's why updating of XML with cron job can be a solution. And such an approach is decreasing response time.
This is initial implementation with base functionality, but then we can add more features in the future:
* Write file with small parts to decrease memory usage.
* Refresh of the sitemap.xml file from admin UI.
* Split the sitemap.xml file to several files if the number of links is more than 50k (or more than 10M).

Steps for review:
* Install BigTree CMS
* Add several pages.
* Go to `/sitemap.xml` and make sure page looks ok.
* Create new pages.
* Execute BigTree cron.
* Go to `/sitemap.xml` and make sure new links were added to the sitemap.xml file.
